src/coreComponents/physicsSolvers/solidMechanics/kernels/ImplicitSmallStrainQuasiStatic_impl.hpp

```diff
@@ -0,0 +1,190 @@
+                                   m_gravityVector[2] * m_density( k, q )* detJ };
+
+  real64 N[numNodesPerElem];
+  FE_TYPE::calcN( q, N );
+  FE_TYPE::plusGradNajAijPlusNaFi( dNdX,
+                                   stress,
+                                   N,
+                                   gravityForce,
+                                   reinterpret_cast< real64 (&)[numNodesPerElem][3] >(stack.localResidual) );
+  stiffness.template upperBTDB< numNodesPerElem >( dNdX, -detJ, stack.localJacobian );
+}
+
+
+template< typename SUBREGION_TYPE,
+          typename CONSTITUTIVE_TYPE,
+          typename FE_TYPE >
+GEOSX_HOST_DEVICE
+GEOSX_FORCE_INLINE
+real64 ImplicitSmallStrainQuasiStatic< SUBREGION_TYPE, CONSTITUTIVE_TYPE, FE_TYPE >::complete( localIndex const k,
+                                                                                               StackVariables & stack ) const
+{
+  GEOSX_UNUSED_VAR( k );
+  real64 maxForce = 0;
+
+  // TODO: Does this work if BTDB is non-symmetric?
+  CONSTITUTIVE_TYPE::KernelWrapper::DiscretizationOps::template fillLowerBTDB< numNodesPerElem >( stack.localJacobian );
+
+  for( int localNode = 0; localNode < numNodesPerElem; ++localNode )
+  {
+    for( int dim = 0; dim < numDofPerTestSupportPoint; ++dim )
+    {
+      localIndex const dof =
+        LvArray::integerConversion< localIndex >( stack.localRowDofIndex[ numDofPerTestSupportPoint * localNode + dim ] - m_dofRankOffset );
+      if( dof < 0 || dof >= m_matrix.numRows() )
+        continue;
+      m_matrix.template addToRowBinarySearchUnsorted< parallelDeviceAtomic >( dof,
+                                                                              stack.localRowDofIndex,
+                                                                              stack.localJacobian[ numDofPerTestSupportPoint * localNode + dim ],
+                                                                              numNodesPerElem * numDofPerTrialSupportPoint );
+
+      RAJA::atomicAdd< parallelDeviceAtomic >( &m_rhs[ dof ], stack.localResidual[ numDofPerTestSupportPoint * localNode + dim ] );
+      maxForce = fmax( maxForce, fabs( stack.localResidual[ numDofPerTestSupportPoint * localNode + dim ] ) );
+    }
+  }
+
+
+  return maxForce;
+}
+
+template< typename SUBREGION_TYPE,
+          typename CONSTITUTIVE_TYPE,
+          typename FE_TYPE >
+template< typename POLICY,
+          typename KERNEL_TYPE >
+real64
+ImplicitSmallStrainQuasiStatic< SUBREGION_TYPE, CONSTITUTIVE_TYPE, FE_TYPE >::kernelLaunch( localIndex const numElems,
+                                                                                            KERNEL_TYPE const & kernelComponent )
+{
+  return Base::template kernelLaunch< POLICY, KERNEL_TYPE >( numElems, kernelComponent );
+}
+
+
+} // namespace solidMechanicsLagrangianFEMKernels
+
+} // namespace geosx
+
+#include "finiteElement/kernelInterface/SparsityKernelBase.hpp"
```

Comment:
Do we need to include the file here too?

Reply:
Sorry. Which " the file" are you referring to?

Reply:
I'm not sure that this include is required
```suggestion
```

Reply:
correct